docs/developer_guide.md

```diff
@@ -0,0 +1,59 @@
+# Setup Envoy AI Gateway Local Development Environment
+
+## Prerequisites
+- Go 1.23 or later
```

Comment:
Let's not pin Go version one year later this could be wrong and we always build latest Go.

```suggestion
- The latest Go
```